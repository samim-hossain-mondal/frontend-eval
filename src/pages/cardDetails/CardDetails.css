.card-details{
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    padding: 20px;
    margin-bottom: 100px;
}

.card-d{
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 40%;
    height: 40%;
    background-color: black;
    color: white;
    border-radius: 10px;
    box-shadow: 0 0 10px rgba(0,0,0,0.1);
}

.card-image-d{
    width:100%;
    height:250px;
}

.card-image-d > img{
    width:100%;
    height:100%;
    object-fit:cover;
    border-radius: 10px;
}
.card-content-d{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    height: 100%;
    padding: 1rem;
}

.card-footer-d{
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 1rem;
}

.bookmark-btn-d > button {
    background-color: transparent;
    background-repeat: no-repeat;
    border: none;
    cursor: pointer;
    outline: none;
}

.registration-d{
    display:flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
}

.registration-d > p{
    color: #A0F3AD;
    margin-left: 7px;
}

.seats-d{
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
}

.seats-d > p{
    color: #ECECAB;
    margin-left: 7px;
}

.register-unregister{
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
}

.register-unregister > button{
    color: black;
    padding: 10px;
    margin-bottom: 20px;
    border-radius: 5px;
    border: none;
    cursor: pointer;
}