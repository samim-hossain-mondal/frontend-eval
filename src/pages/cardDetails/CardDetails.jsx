import React,{useState, useEffect} from 'react';
import { useParams } from 'react-router';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faBookmark as faBookmarkRegular } from '@fortawesome/free-regular-svg-icons';
import { faBookmark as faBookmarkSolid} from '@fortawesome/free-solid-svg-icons';
import { faCircleCheck } from '@fortawesome/free-solid-svg-icons';
import { faCircleXmark } from '@fortawesome/free-solid-svg-icons';
import {BASE_URL, API_URL} from '../../constants/url';
import './CardDetails.css';

export default function CardDetails() {
  const { id } = useParams();
  const [event, setEvent] = useState({});
  const fetchEvent = async () => {
    const response = await fetch(`${BASE_URL}/${API_URL}/${id}`);
    const data = await response.json();
    setEvent(data);
  };
  useEffect(() => {
    fetchEvent();
  }, []);
  return (
    <div className='card-details'>
      <div className="card-d">
        <div className="card-image-d">
          <img src={event.imgUrl} alt="event" />
          <hr/>
        </div>
        <div className="card-content-d">
          <div className="card-title-d">
            <h3>{event.name}</h3>
          </div>
          <div className="card-description-d">
            <p>{event.description}</p>
          </div>
          <div className="card-venue-d">
            <div>
              <a>VENUE: {event.venue}</a>
            </div>
            <div>
              <a>DATE: {event.datetime}</a>
            </div>
          </div>
        </div>
        <div className='card-footer-d'>
          <div>
            {
              event.isRegistered ? 
                (
                  <div className='registration-d'>
                    <FontAwesomeIcon icon= {faCircleCheck} fontSize='25' color='#A0F3AD' /> 
                    <p>REGISTERED</p>
                  </div>
                ) 
                : event.areSeatsAvailable ? 
                  (
                    <p></p>
                  ) 
                  : 
                  (
                    <div className='seats-d'>
                      <FontAwesomeIcon icon= {faCircleXmark} fontSize='25' color='#ECECAB'/>
                      <p>NO SEATS AVAILABLE</p>
                    </div>
                  )
            }
          </div>
          <div className='bookmark-btn-d'>
            <button>
              {
                event.isBookmarked ? 
                  <FontAwesomeIcon icon={faBookmarkSolid} color='#EA8282' fontSize='25'/> 
                  : 
                  <FontAwesomeIcon icon={faBookmarkRegular} color='#EA8282' fontSize='25'/>
              }
            </button>
          </div>
        </div>
        <div className='register-unregister'>
          {
            event.isRegistered ?
              ( 
                <button className='cancel-btn'><b>UNREGISTER</b></button>
              )
              :
              (
                event.areSeatsAvailable ?

                  (
                    <button className='register-btn'><b>REGISTER</b></button>
                  )
                  :
                  (
                    <a></a>
                  )
              )
          }
        </div>
      </div>
    </div>
  );
}