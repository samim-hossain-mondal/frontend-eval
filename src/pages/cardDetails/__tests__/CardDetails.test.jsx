import React from 'react';
import { render } from '@testing-library/react';
import CardDetails from '../CardDetails';

describe('CardDetails', () => {
  it('should render correctly', () => {
    const { container } = render(<CardDetails />);
    expect(container).toMatchSnapshot();
  });
}
);