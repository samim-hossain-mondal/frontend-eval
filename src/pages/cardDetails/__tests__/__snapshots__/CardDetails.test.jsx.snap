// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDetails should render correctly 1`] = `
<div>
  <div
    class="card-details"
  >
    <div
      class="card-d"
    >
      <div
        class="card-image-d"
      >
        <img
          alt="event"
        />
        <hr />
      </div>
      <div
        class="card-content-d"
      >
        <div
          class="card-title-d"
        >
          <h3 />
        </div>
        <div
          class="card-description-d"
        >
          <p />
        </div>
        <div
          class="card-venue-d"
        >
          <div>
            <a>
              VENUE: 
            </a>
          </div>
          <div>
            <a>
              DATE: 
            </a>
          </div>
        </div>
      </div>
      <div
        class="card-footer-d"
      >
        <div>
          <div
            class="seats-d"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-circle-xmark "
              color="#ECECAB"
              data-icon="circle-xmark"
              data-prefix="fas"
              focusable="false"
              font-size="25"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM175 175c9.4-9.4 24.6-9.4 33.9 0l47 47 47-47c9.4-9.4 24.6-9.4 33.9 0s9.4 24.6 0 33.9l-47 47 47 47c9.4 9.4 9.4 24.6 0 33.9s-24.6 9.4-33.9 0l-47-47-47 47c-9.4 9.4-24.6 9.4-33.9 0s-9.4-24.6 0-33.9l47-47-47-47c-9.4-9.4-9.4-24.6 0-33.9z"
                fill="currentColor"
              />
            </svg>
            <p>
              NO SEATS AVAILABLE
            </p>
          </div>
        </div>
        <div
          class="bookmark-btn-d"
        >
          <button>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-bookmark "
              color="#EA8282"
              data-icon="bookmark"
              data-prefix="far"
              focusable="false"
              font-size="25"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 48C0 21.5 21.5 0 48 0l0 48V441.4l130.1-92.9c8.3-6 19.6-6 27.9 0L336 441.4V48H48V0H336c26.5 0 48 21.5 48 48V488c0 9-5 17.2-13 21.3s-17.6 3.4-24.9-1.8L192 397.5 37.9 507.5c-7.3 5.2-16.9 5.9-24.9 1.8S0 497 0 488V48z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="register-unregister"
      >
        <a />
      </div>
    </div>
  </div>
</div>
`;
